### PR TITLE
Cleaning todos

### PR DESCRIPTION
**Pull request by pmerku**
# todos done
- [x] parser
- [x] directory traversal detection
- [x] use doWrite
- [x] remove localtime todo 
- [x] route body limit
- [x] cgi envs
- [x] cgi 404 handling
- [x] possible race condition

### CODE REVIEW

File: src/server/http/HTTPParser.cpp [hunk 1]
```diff
@@ -140,12 +140,16 @@ HTTPParser::ParseReturn		HTTPParser::parseHeaders(HTTPParseData &data, const std
...
+			std::vector<std::string> headerValue = utils::split(value, ",");
+			for (std::vector<std::string>::iterator valueIt = headerValue.begin(); valueIt != headerValue.end(); ++valueIt) {
+				std::string::size_type start = (*valueIt).find_first_not_of(' ');
+				std::string::size_type end = (*valueIt).find_last_not_of(' ');
+				*valueIt = (*valueIt).substr(start, end+1 - start);
+				if ((*valueIt).find("chunked") == std::string::npos) {
+					globalLogger.logItem(logger::ERROR, "Not supported transfer encoding");
+					data.parseStatusCode = 400;
+					return ERROR;
+				}
```
Comment (mrjvs) [inline src/server/http/HTTPParser.cpp]:
```
if (*valueIt == "chunked")
  _isChunked = true;
else {
  globalLogger.logItem(logger::ERROR, "Not supported transfer encoding");
					data.parseStatusCode = 400;
					return ERROR;
}
```

File: src/utils/Uri.cpp [hunk 2]
```diff
@@ -25,9 +25,19 @@ Uri::Uri(const std::string &in): path(), query(), identifier() {
...
+	// check if path is '.' or '..' and erase it
+	std::string::size_type dot = p.find('.');
+	std::string::size_type dots = p.find("..");
+	if (dot == 0) {
+		p.erase(dot, 1);
+	}
+	if (dots == 0) {
+		p.erase(dots, 2);
+	}
+
```
Comment (mrjvs) [inline src/utils/Uri.cpp]: make seperate (static function or bool in parameter) check that only applies in the parser. directory traversal is used in config options.

Comment (mrjvs) [reply src/utils/Uri.cpp]: (also check with telnet and nginx if it should be 400'ed or erased)

**Review by mrjvs — CHANGES_REQUESTED** [review]
also try doRead again with CGI

**Review by mrjvs — APPROVED** [review]

**Review by maartenRamadan — APPROVED** [review]

**Pull request merged**